File: tools-manage/invocationlab-admin/src/main/java/org/javamaster/invocationlab/admin/service/Pair.java
package org.javamaster.invocationlab.admin.service;

import lombok.Getter;

import java.util.StringJoiner;

/**
 * @author yudong
 */
@Getter
public class Pair<L, R> {

    private final L left;
    private final R right;

    public Pair(L left, R right) {
        this.left = left;
        this.right = right;
    }

    public static <L, R> Pair<L, R> of(L left, R right) {
        return new Pair<>(left, right);
    }

    @Override
    public String toString() {
        return new StringJoiner(", ", Pair.class.getSimpleName() + "[", "]")
                .add("left=" + left)
                .add("right=" + right)
                .toString();
    }

}

File: tools-manage/invocationlab-admin-21/src/main/kotlin/org/javamaster/invocationlab/admin/service/Pair.kt
package org.javamaster.invocationlab.admin.service

import java.util.*

/**
 * @author yudong
 */
class Pair<L, R>(val left: L, val right: R) {
    override fun toString(): String {
        return StringJoiner(", ", Pair::class.java.simpleName + "[", "]")
            .add("left=$left")
            .add("right=$right")
            .toString()
    }

    companion object {
        fun <L, R> of(left: L, right: R): Pair<L, R> {
            return Pair(left, right)
        }
    }
}
